### PR TITLE
Added gulp and eslint

### PR DESCRIPTION
**Pull request by Teodor92**
Added gulp and eslint to the project. Refactored all the JavaScript files to comply with the ESLint code style.

How to run the ESLint linter?
1. Install Node.JS on your machine.
2. In the OJS.Web project run the following command:
`npm install`
3. You can now use the linter as a gulp task from the Task Runner Explorer in Visual Studio.

### CODE REVIEW

File: Open Judge System/Web/OJS.Web/Scripts/Administration/Tests/tests-index.js [hunk 1]
```diff
@@ -78,30 +96,30 @@ function populateDropDowns(problemIdAsString) {
...
-
-        var problemsData = new kendo.data.DataSource({
-            transport: {
-                read: {
-                    url: '/Administration/Tests/GetCascadeProblems/' + contestId.toString(),
-                    dataType: 'json'
-                }
-            }
-        });
+        ////var contestsData = new kendo.data.DataSource({
```
Comment (jicata) [inline Open Judge System/Web/OJS.Web/Scripts/Administration/Tests/tests-index.js]: Are these functions to be deleted? If so you can just remove them.

File: Open Judge System/Web/OJS.Web/Scripts/Administration/Contests/contests-index.js [hunk 2]
```diff
@@ -1,13 +1,25 @@
-﻿function getTypeName(type) {
+﻿/* exported getTypeName */
```
Comment (jicata) [inline Open Judge System/Web/OJS.Web/Scripts/Administration/Contests/contests-index.js]: Does this (and all the rest of the "exported" comments) mean "TODO"?

Comment (Teodor92) [inline Open Judge System/Web/OJS.Web/Scripts/Administration/Contests/contests-index.js]: The exported comments are used to indicated which functions are public, for the specific file, so the linter can mark them as "exported" (ES5 has no way to "export" stuff).

See this for more info: 
http://eslint.org/docs/rules/no-unused-vars#exported

File: Open Judge System/Web/OJS.Web/Scripts/Administration/Tests/tests-index.js (hunk 1 above)
Comment (Teodor92) [inline Open Judge System/Web/OJS.Web/Scripts/Administration/Tests/tests-index.js]: Removed.

**Comment by KristianMariyanov** [comment]
Why do you close the PR? 🤔

**Pull request merged**